src/main/java/com/google/sps/authentication/UserAuthenticationStatus.java

```diff
@@ -0,0 +1,74 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.authentication;
+
+import org.jetbrains.annotations.Nullable;
+import java.lang.IllegalStateException;
+
+/** Class containing user's authentication status. */
+public class UserAuthenticationStatus {
+
+    private final boolean isLoggedIn;
+
+    @Nullable
+    private final String loginUrl, logoutUrl;
+
+    private UserAuthenticationStatus(boolean isLoggedIn, String loginUrl, String logoutUrl) {
+        this.isLoggedIn = isLoggedIn;
+        this.loginUrl = loginUrl;
+        this.logoutUrl = logoutUrl;
+    }
+
+    public static class Builder {
+        private final boolean isLoggedIn;
+        private String loginUrl, logoutUrl;
+        public Builder(boolean isLoggedIn) {
+            this.isLoggedIn = isLoggedIn;
+        }
+        public Builder setLoginUrl(String loginUrl) {
+            if (this.isLoggedIn) {
+                throw new IllegalStateException(
+                            "User is already logged in, failed to create a login url!"); 
+            }
```

Comment:
Move this logic to build() instead. It simpler and more readable.

Then isLoggedIn doesn't need to be final, it can have a setter.

Reply:
I removed the Builder, but I kept the isLoggedIn since i need it for the client.